src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import duke.storage.Storage;
+import duke.storage.StorageException;
+import duke.task.Task;
+import duke.task.TaskException;
+import duke.task.TaskList;
+import duke.task.TaskType;
+
+/**
+ * Encapsulates the logic for adding tasks.
+ */
+public class AddCommand extends Command {
+    private final TaskType taskType;
+
+    /**
+     * Constructs a AddCommand with a given TaskType.
+     *
+     * @param args Arguments for the command.
+     * @param taskType TaskType for which task to add.
+     */
+    public AddCommand(String args, TaskType taskType) {
+        super(args);
+        this.taskType = taskType;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage) throws TaskException, StorageException {
```

Comment:
Should Javadoc comments be added to this public method?

Reply:
Nice catch! This is quite interesting - Javadocs for an overridden method. I've added Javadocs for `AddCommand.class` only since it does have some logic that isn't immediately apparent. I'm not very sure about the rest though, since the inherited classes will be able to 'copy' over their parent class' Javadocs.